Ignore stale notification queues from notification.xml

Praveenrajmani opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
If the ARN does not exist in the targetlist, remove it from the notification.xml.

## Motivation and Context
Fixes #7650

## Regression
No

## How Has This Been Tested?
The steps are provided in #7650 

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [x] All new and existing tests passed.

sinhaashish reviewed (commented):
Tested and its working . But IMO User should be prompted about the stale entries, before removing.

Praveenrajmani committed:
Ignore stale notification queues in notification.xml

Allow renaming/editing a notification config. By replying with a successful GetBucketNotification
response, without checking for any missing config ARN in targetList.

Fixes #7650

balamurugana commented on cmd/bucket-notification-handlers.go:
```diff
@@ -71,30 +72,21 @@ func (api objectAPIHandlers) GetBucketNotificationHandler(w http.ResponseWriter,
+			writeErrorResponse(ctx, w, toAPIError(ctx, err), r.URL, guessIsBrowserReq(r))
+			return
+		}
+		configData, err = xml.Marshal(&event.Config{})
```
you would need to set `XMLNS`

Praveenrajmani replied:
done :+1:

balamurugana commented on cmd/bucket-notification-handlers.go:
```diff
@@ -142,9 +134,10 @@ func (api objectAPIHandlers) PutBucketNotificationHandler(w http.ResponseWriter,
-
-		writeErrorResponse(ctx, w, apiErr, r.URL, guessIsBrowserReq(r))
-		return
+		if _, ok := err.(*event.ErrARNNotFound); !ok {
```
Why is unavailable ARN supported for `Put Bucket Notification`?

Praveenrajmani replied:
Previously, we would not save the notification config even if one of the target ARNs is not found. By ignoring this ARN not found error we would save the remaining ARNs atleast in the config

balamurugana commented on pkg/event/config.go:
```diff
@@ -272,15 +272,14 @@ func (conf *Config) ToRulesMap() RulesMap {
 
-	return &config, nil
+	config.SetRegion(region)
+	return &config, err
```
What does it mean to return non-nil config and error?

Praveenrajmani replied:
Now that we should not send an nil config for an `ARN Not Found` error. the config will not be empty/nil any more. By this way we would always save the config even if one of the ARN is not found

sinhaashish reviewed (approved):
LGTM
